keras/engine/training.py

```diff
@@ -258,7 +259,14 @@ def collect_trainable_weights(layer):
         weights += layer.trainable_weights
     # dedupe weights
     weights = list(set(weights))
-    weights.sort(key=lambda x: x.name)
+    # TF variables have auto-generated the name, while Theano has auto-generated the auto_name variable. name in Theano is None
+    if weights:
```

Comment:
This still seems hackey. Maybe a better or more clear situation could be something like:

```python
from keras import backend as K
if K._BACKEND == 'tensorflow':
   logic
else:
   other_log
```